File: src/src_core/fmg/core/mosaic/draw/ICellPaint.cs
using fmg.core.mosaic.cells;

namespace fmg.core.mosaic.draw {

   /// <summary>
   /// Interface for drawing cell
   /// </summary>
   public interface ICellPaint<in TPaintable, TImage, TPaintContext>
      where TPaintable : IPaintable
      where TImage : class
      where TPaintContext : PaintCellContext<TImage>
   {
      TPaintContext PaintContext { get; set; }

      void Paint(BaseCell cell, TPaintable paint);

      void PaintBorder(BaseCell cell, TPaintable paint);

      /// <summary> draw border lines </summary>
      void PaintBorderLines(BaseCell cell, TPaintable paint);

      void PaintComponent(BaseCell cell, TPaintable paint);

      /// <summary> залить ячейку нужным цветом </summary>
      void PaintComponentBackground(BaseCell cell, TPaintable paint);

   }

}

File: src/src_core/fmg/core/mosaic/draw/ICellPaint.java
package fmg.core.mosaic.draw;

import fmg.core.mosaic.cells.BaseCell;

/**
 * Interface for drawing cell
 *
 * @param <TPaintable> see {@link IPaintable}
 * @param <TImage> plaform specific view/image/picture or other display context/canvas/window/panel
 * @param <TPaintContext> see {@link MosaicDrawModel}
 */
public interface ICellPaint<TPaintable extends IPaintable, TImage, TPaintContext extends MosaicDrawModel<TImage>> {

   void paint(BaseCell cell, TPaintable p, TPaintContext paintContext);

}
